### PR TITLE
Make options optional on child()

### PR DESCRIPTION
**Pull request by kevincasey1222**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
 export interface HierarchicalTokenBucketOptions {
   /**
    * The total number of requests allowed when the bucket is full.
+   * maximumCapacity should be >= 1.
```
Comment (ndowmon) [inline src/index.ts:4]: Thanks for the docs improvement 💯

File: src/index.ts [hunk 2]
```diff
@@ -56,9 +58,10 @@ export class HierarchicalTokenBucket {
     this.capacity = options.maximumCapacity;
   }
 
-  child(options: Omit<HierarchicalTokenBucketOptions, 'parent'>) {
+  child(options?: Omit<HierarchicalTokenBucketOptions, 'parent'>) {
```
Comment (ndowmon) [inline src/index.ts:61]: Would it make sense to allow users to specify just one or the other? 

```ts
const childBucket = parentBucket.child({ refillRate: 10 });
```

```suggestion
  child(options?: Omit<Optional<HierarchicalTokenBucketOptions>, 'parent'>) {
```

**Review by ndowmon — APPROVED** [review]
LGTM, but added one question about the interface

File: src/index.ts (hunk 2 above)
Comment (kevincasey1222) [reply src/index.ts:61]: If I do that, I have to make the properties of `HierarchicalTokenBucketOptions` optional, and then check for undefined values in the constructor... I think it's messy and not a common use case. I figure the use case of inheriting options from the parent makes sense, but if a person is going to change an option anyway, it's not that painful to specify both options in that case.

Kinda related, I could totally see a future PR with setters for `maximumCapacity` and `refillRate`, and then using those when throttling errors are received to make a "smart bucket" that modulates itself.

Comment (ndowmon) [reply src/index.ts:61]: I love the vision @kevincasey1222

**Pull request merged**